include/s4.h

```diff
@@ -0,0 +1,325 @@
+/**
+ * @file s4.h
+ *
+ * Copyright (C) Metaswitch Networks 2018
+ * If license terms are provided to you in a COPYING file in the root directory
+ * of the source code repository by which you are accessing this code, then
+ * the license outlined in that COPYING file applies to your use.
+ * Otherwise no rights are granted except for those provided to you by
+ * Metaswitch Networks in a separate written agreement.
+ */
+
+
+#ifndef S4_H__
+#define S4_H__
+
+#include <string>
+#include <list>
+#include <map>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "sas.h"
+#include "associated_uris.h"
+#include "astaire_aor_store.h"
+#include "httpclient.h"
+#include "chronosconnection.h"
+
+class S4
+{
+public:
+  /// @class S4::ChronosTimerRequestSender
+  ///
+  /// Class responsible for sending any requests to Chronos about
+  /// registration/subscription expiry
+  ///
+  /// @param chronos_conn    The underlying chronos connection
+  class ChronosTimerRequestSender
+  {
+  public:
+    ChronosTimerRequestSender(ChronosConnection* chronos_conn);
+
+    virtual ~ChronosTimerRequestSender();
+
+    /// Create and send any appropriate Chronos requests
+    ///
+    /// @param aor_id       The AoR ID
+    /// @param aor_pair     The AoR pair to send Chronos requests for
+    /// @param now          The current time
```

Comment:
It can, but then it will use a slightly different time than the memcached store does for working out expiry. It makes sense to me to have S4 work out the time, then pass in a consistent time to everything that uses it.

Reply:
Ah, so it sounds like this parameter isn't really the "current time" then? Should this function not just take an expiry time actually?

I agree to a certain extent that having S4 work out something consistent is nice, but calling it `current_time` doesn't reflect that at all and is misleading

Reply:
Yup, I agree that the parameter name should be clearer.